### DISCUSSION TITLE
Full stack "native" port of xMicroG(Java) + VManager(Kotlin) + Integrations(Java) + Vanced(Py/Sh) to Linux/Mac/Win

### DISCUSSION THREAD

**Discussion by J-HS-L** [TeamVanced/VancedManager#796] [Ideas]
Hi folks, as the title suggests....

Want to contribute towards building something truly native for GNU/*nix, macOS, & Win. 10/11. A suite that's; 
Highly optimised/performant/stable, with UX's that behave/feel/look like they're "Inbox Apps" on each platform.

All using entirely or mostly, the same code-base...

Some of what's already used might be fine to re-use, but it's likely that a sizable chunk must be refactored or ported;
As there is only a handful of optimal languages/libraries/tooling, for such an undertaking. 

Is anyone at YTVanced or related F/OSS communities actually doing something like this; 
Or have they at least begun some preliminary planning towards a similar goal???

Thank you very much,
SJ

**Comment by X1nto** [comment]
Hi, thanks for your interest in the project. I want to clarify, do you mean porting Vanced Manager or the YouTube app itself? The latter is rather impossible while porting Manager doesn't really make sense (unless, maybe, you want to install apps on your device "remotely"?)

**Comment by X1nto** [comment]
Oops, seems like I forgot to read the title. Never mind the last question. Are there any reasons you'd want to port Android apps to desktop platforms? IMO, It's better to make apps for them from the ground up. For Windows with C#, for macOS with Swift, and for Linux with C/C++. Although, it would be easier to use Kotlin Multiplatform with Jetpack Compose + platform-specific UI components.

**Comment by J-HS-L** [comment]
Hi, thank-you for getting back to me RE this.

As mentioned, not necessarily "port"... 
It really depends: on my findings, and how everything evolves.

First want to be sure there's not something like this in the community already.

If there is....
I'd prefer to join that effort, or start a complimentary codebase(s). 

~10pm, bed time, TY/BR.

**Reply by KevinX8** [reply]
I would say the closest thing to what you're looking to make on pc is this: https://invidious.io/ but it's more of an alternative web front end than an entirely native pc app

**Reply by Heldenkrieger01** [reply]
Also a combination of an adblocker (uBlock, Brave Browser, etc.) together with the SponsorBlock extension work fine for me.
